Feature | ER-6 Add Prisma and Zod for Database Integration and Validation

Aathil1681 opened the pull request:
## Add Prisma and Zod for Database Integration and Validation

https://kifgo.atlassian.net/browse/ER-6

## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Refactoring
- [ ] Security patch
- [ ] UI/UX improvement

## Description

This PR integrates Prisma as an ORM for database management and Zod for schema validation into the employee-rating project. The changes include updates to the `.gitignore` file to include `.env`, modifications to `package.json` and `package-lock.json` to add `@prisma/client`, `prisma`, and `zod` as dependencies, and reorganization of the dependency lists for consistency.

## What is fixed / implemented?

This PR implements database connectivity using Prisma and adds runtime validation with Zod. It enables structured data management and ensures data integrity through schema validation.

- Disabled no-var rule in ESLint configuration
- Renamed DB_URL to DATABASE_URL in Prisma schema

## Additional Information

Ensure the Prisma schema is defined and migrations are run post-deployment. The `.env` file should be configured with database credentials and not committed to version control.

## Screenshots (if applicable)

_N/A_

## Checklist

- [ ] I have added or updated relevant tests to cover the changes.
- [X] I have performed a self-review of my code.
- [X] My changes generate no new warnings or errors in development and production builds.
- [ ] I have verified the changes on multiple devices and browsers (if applicable).
- [X] Environment variables have been updated (if applicable).
- [X] New packages have been installed and documented (if applicable).

Aathil1681 committed:
chore(deps): add Prisma and Zod dependencies

- Add @prisma/client and prisma for database integration
- Add zod for schema validation
- Update .gitignore to ignore only .env file explicitly

Dhuvii commented on .env.sample:
```diff
@@ -0,0 +1 @@
+DATABASE_URL=<MONGO_DB_URL>
```
Change the value as DB_URL, not MONGO_DB_URL

Aathil1681 committed:
Add Prisma schema and update .env.sample for database configuration

This commit introduces the Prisma schema file (`prisma/schema.prisma`) to define the database structure using MongoDB as the provider. It also updates the `.env.sample` file to rename the `DATABASE_URL` variable to `DB_URL` for consistency with the schema. These changes set up the foundation for database integration in the employee-rating project.

Aathil1681 committed:
feat: update ESLint and Prisma configurations

- Disabled no-var rule in ESLint for flexible variable declarations
- Renamed DB_URL to DATABASE_URL in Prisma schema for consistency

Dhuvii committed:
Merge branch 'staging' into ER-6-install-and-configure-prisma
